orion/eval/evaluation.py

```diff
@@ -0,0 +1,241 @@
+import numpy as np
+from sklearn import metrics
+
+
+class Evaluator:
+    """Evaluator Class.
+
+    The Evaluator Class provides a mechanic to assess the perfomance of
+    anomaly detection methods. This class addresses the problem as
+    contextual anomalies.
+
+    anomaly = [(start, end)] inclusive interval.
+
+    """
+
+    def _overlap(self, expected, observed):
+        first = expected[0] - observed[1]
+        second = expected[1] - observed[0]
+        return first * second < 0
+
+    def _any_overlap(self, part, intervals):
+        for interval in intervals:
+            if self._overlap(part, interval):
+                return 1
+
+        return 0
+
+    def _pad(self, lst):
+        return [(part[0], part[1] + 1) for part in lst]
+
+    def _partition(self, expected, observed, start=None, end=None):
+        edges = set()
+
+        if start is not None:
+            edges.add(start)
+
+        if end is not None:
+            edges.add(end)
+
+        for edge in expected + observed:
+
+            edges.update(edge)
+
+        partitions = list()
+        edges = sorted(edges)
+        last = edges[0]
+        for edge in edges[1:]:
+            partitions.append((last, edge))
+            last = edge
+
+        expected_parts = list()
+        observed_parts = list()
+        weights = list()
+        for part in partitions:
+            weights.append(part[1] - part[0])
+            expected_parts.append(self._any_overlap(part, expected))
+            observed_parts.append(self._any_overlap(part, observed))
+
+        return expected_parts, observed_parts, weights
+
+    def _weighted_segment(self, expected, observed, start=None, end=None):
+        expected, observed, weights = self._partition(expected, observed, start, end)
+
+        return metrics.confusion_matrix(
+            expected, observed, sample_weight=weights, labels=[0, 1]).ravel()
+
+    def _overlap_segment(self, expected, observed, start=None, end=None):
+        tp, fp, fn = 0, 0, 0
+
+        observed_copy = observed.copy()
+
+        for expected_seq in expected:
+            found = False
+            for observed_seq in observed:
+                if self._overlap(expected_seq, observed_seq):
+                    if not found:
+                        tp += 1
+                        found = True
+                    if observed_seq in observed_copy:
+                        observed_copy.remove(observed_seq)
+            if not found:
```

Comment:
Can we add blank lines before reducing indentation?
This creates a clearer visual separation between blocks.
For example, here I would add them under lines 80 and 82, and optionally under 78 as well.